src/Parser.cpp

```diff
@@ -79,10 +79,12 @@ std::vector<Entry> Parser::parseMeasurements(json measurements_json) const {
     uint64_t stamp = entry_element["stamp"].get<uint64_t>();
     gtsam::NonlinearFactorGraph entry_measurements;
     std::vector<std::string> type_tags;
-    for (auto& measurement : entry_element["measurements"]) {
-      std::string tag = measurement["type"].get<std::string>();
-      type_tags.push_back(tag);
-      entry_measurements.push_back(measurement_parsers_.at(tag)(measurement));
+    if(entry_element.contains("measurements")) {
```

Comment:
Generally, you probably shouldn't have one. 

In my case, I'm using jrl to save a stereo-factor-only dataset since computing those factors is rather intensive, requiring feature detection/matching/ransac. I then take the saved stereo-factor-only dataset and combine it with whatever backbone I'm experimenting with adding to it. The stereo-factor-only dataset occasionally has empty entries due to poor matching (and the initial entry is always empty as well). It's important to include these empty entries to make sure indices line up when adding the odometry backbone.

At minimum there should be some sort of check here - I was able to successfully write jrl files with an empty entry, but reading them in resulted in an obscure json error.

Reply:
Ah okay, if the library can write it, we should support reading it even if empty! And I think im okay with a more permissible behavior than a more restrictive one, since these are going to be edge case situations anyways.